Makefile

```diff
@@ -1,162 +1,217 @@
+FAKE_CONTENT := "This is a fake file generated by 'make install'"
+
+NODE_MODULES := frontend/node_modules
+PACKAGE_JSON := frontend/package.json
+NODE_MODULES_NEWER := $(shell [ ! -e $(NODE_MODULES) ] || [ $(NODE_MODULES) -ot $(PACKAGE_JSON) ]; echo $$?)
+E2E_VENV := e2e/venv
+E2E_INSTALL := e2e/install_sh
+E2E_REQUIREMENTS := e2e/requirements.txt
+E2E_INSTALL_NEWER := $(shell [ ! -e "$(E2E_VENV)" ] || [ "$(E2E_VENV)" -ot "$(E2E_INSTALL)" ] || [ "$(E2E_VENV)" -ot "$(E2E_REQUIREMENTS)" ]; echo $$?)
+
 .PHONY: all
 all: build
 
+
 .PHONY: help
-help:  ## Show help for each of the Makefile recipes.
+help:
 	@awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z0-9_-]+:.*?## / {gsub("\\\\n",sprintf("\n%22c",""), $$2);printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST)
 
+
+# Runs (Implicit builds)
+
+
 .PHONY: run-frontend
-run-frontend: build-frontend  ## Build & Run frontend.
-	cd frontend && npm run dev -- --host
+run-frontend: build-frontend  ## Build & Run frontend
+	@cd frontend && npm run dev -- --host
 
 .PHONY: run-backend
-run-backend: build-backend  ## Build & Run backend.
-	cd backend && make run
+run-backend: build-backend  ## Build & Run backend
+	@cd backend && make run
 
 .PHONY: run-reset-backend
-run-reset-backend: build-backend reset-database  ## Build & Run backend with a database reset.
-	cd backend && make run
+run-reset-backend: reset-database  ## Build & Run backend with a database reset
+	$(MAKE) run-backend
 
 .PHONY: run-mdbook
-run-mdbook: build-mdbook  ## Build & Run mdbook.
-	mdbook serve --open
+run-mdbook: build-mdbook  ## Build & Run mdbook
+	@mdbook serve --open
 
 .PHONY: run-storybook
-run-storybook: build-storybook  ## Build & Run storybook.
-	cd frontend && npm run storybook
+run-storybook: build-storybook  ## Build & Run storybook
+	@cd frontend && npm run storybook
+
+
+# TEST (Implicit builds)
 
-# TEST
 
 .PHONY: test
-test: test-frontend test-backend test-mdbook test-e2e  ## Test everything.
-	pre-commit
+test: test-frontend test-backend test-mdbook test-e2e  ## Test everything
+	$(MAKE) pre-commit
 
 .PHONY: test-e2e
-test-e2e: ## End-to-End tests. Needs install-e2e, backend and frontend running.
-	cd e2e && ./e2e.sh
+test-e2e:  ## End-to-End tests. Needs install-e2e, backend and frontend running
+	@cd e2e && ./e2e.sh
 
 .PHONY: test-frontend
-test-frontend:  ## Test Frontend.
-	cd frontend && npm install && npm run format:check && npm run lint && npm run test
+test-frontend:  build-frontend  ## Build & Test Frontend
+	@cd frontend && npm install && npm run format:check && npm run lint && npm run test
 
 .PHONY: test-backend
-test-backend:  ## Test Backend.
-	cd backend && make test
+test-backend:  build-backend  ## Build & Test Backend
+	@make test -C ./backend
 
 .PHONY: test-mdbook
-test-mdbook:  ## Test Mdbook.
-	mdbook test
+test-mdbook:  build-mdbook  ## Build & Test Mdbook
+	@mdbook test
+
+# Test all makefile targets except the ones with "run-".
+.PHONY: test-makefile
+test-makefile: $(MAKEFILE_LIST)
```

Comment:
This is just for testing the makefile itself, should probably remove the phony.

Reply:
It basically executes everything but what starts with `run-`? What it does additionally, so that we can call it test? (I only refer about the documentation, the target itself is okay if you want/need it :wink:)